spog/api/src/endpoints/package.rs

```diff
@@ -84,46 +93,63 @@ pub async fn package_search(
+pub async fn package_get(
+    guac: web::Data<GuacService>,
+    v11y: web::Data<V11yService>,
+    path: web::Path<String>,
+) -> Result<HttpResponse, Error> {
     let purl = path.into_inner();
-    let vex_results = guac.certify_vex(&purl).await?;
-    let mut vex = vex_results
-        .iter()
-        .flat_map(|vex| {
-            vex.vulnerability
-                .vulnerability_ids
-                .iter()
-                .map(|id| V11yRef {
-                    cve: id.vulnerability_id.clone().to_uppercase(),
-                    severity: "unknown".to_string(),
-                })
-                .collect::<Vec<V11yRef>>()
-        })
-        .collect::<Vec<V11yRef>>();
-
-    let vuln_results = guac.certify_vuln(&purl).await?;
-    let mut vulns = vuln_results
-        .iter()
-        .flat_map(|vuln| {
-            vuln.vulnerability
-                .vulnerability_ids
-                .iter()
-                .map(|id| V11yRef {
-                    cve: id.vulnerability_id.clone().to_uppercase(),
-                    severity: "unknown".to_string(),
-                })
-                .collect::<Vec<V11yRef>>()
-        })
-        .collect::<Vec<V11yRef>>();
-    vulns.append(&mut vex);
-
-    let pkg = PackageInfo {
-        purl,
-        vulnerabilities: vulns,
-    };
+    let vulnerabilities = get_vulnerabilities(&guac, &v11y, &purl).await?;
+    let pkg = PackageInfo { purl, vulnerabilities };
     Ok(HttpResponse::Ok().json(&pkg))
 }
 
+/// Retrieve all vulnerabilities (from VEX and CVE) related to the input purl
+#[instrument(skip(guac, v11y), err, ret(level = Level::DEBUG))]
+async fn get_vulnerabilities(
+    guac: &web::Data<GuacService>,
+    v11y: &web::Data<V11yService>,
+    purl: &str,
+) -> Result<Vec<V11yRef>, Error> {
+    let vex_results = guac.certify_vex(purl).await?;
+    let mut vulnerabilities: Vec<V11yRef> = vec![];
+    for certify_vex in vex_results {
+        for vulnerability_id in certify_vex.vulnerability.vulnerability_ids {
+            vulnerabilities.push(get_vulnerability(v11y, vulnerability_id).await?);
+        }
+    }
+    let vuln_results = guac.certify_vuln(purl).await?;
+    for certify_vuln in vuln_results {
+        for vulnerability_id in certify_vuln.vulnerability.vulnerability_ids {
+            vulnerabilities.push(get_vulnerability(v11y, vulnerability_id).await?);
+        }
+    }
+    Ok(vulnerabilities)
+}
+
+/// Retrieve vulnerability information from V11y
+#[instrument(skip(v11y), err, ret(level = Level::DEBUG))]
+async fn get_vulnerability(v11y: &web::Data<V11yService>, vulnerability_id: VulnerabilityId) -> Result<V11yRef, Error> {
+    let cve = vulnerability_id.vulnerability_id.clone().to_uppercase();
+    let severity = Score::from(get_vulnerability_score(v11y, &cve).await?)
+        .severity()
+        .to_string();
+    Ok(V11yRef { cve, severity })
+}
+
+/// Retrieve vulnerability score from V11y
+#[instrument(skip(v11y), err)]
```

Comment:
do all the instrument instructionss need to be at DEBUG level ?

Reply:
Well, usually we keep them at INFO level in all of the other, already instrumented, methods.
I've added some `ret(level = Level::DEBUG)` so that, the verbose output logged, is printed only at DEBUG level.